cpp/modules/math.gl/core/src/core.h

```diff
@@ -123,12 +125,16 @@ auto operator!=(const Vector2<coord> &v1, const Vector2<coord> &v2) -> bool {
 ///////////////////////////////////////////////////////////
 //  Vector3
 
+// TODO(ilija@unfolded.ai): When using std140 memory layout in GLSL vec3 takes in 16bytes
+// We should either use vec3 in our shaders instead, or create a GPUVector3 class that we'd use instead
 template <typename coord>
-class Vector3 {
+class alignas(16) Vector3 {
```

Comment:
This seems like it could cause problems when we allocate Vector3 arrays for attributes.... Can we add the alignas specifier when we declare fields in the uniform struct instead?

Reply:
I tried using `alignas` specified for fields in struct definition, it seems to align the entire structure as seen in the example [here](https://en.cppreference.com/w/c/language/_Alignas). It looks like it can be used with variables, which would make current code a mess.

Added `UniformVector3` for this purpose for now